### PR TITLE
Adding time series

### PR DESCRIPTION
**Pull request by narennadig**
Added Time-series example using holt-winter's algorithm.

### CODE REVIEW

**Comment by narennadig** [comment]
@googlebot I signed it!

File: docs/samples/custom/timeseries-holtwinter/model.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        dict1["predicted_values_holt"]=str(n)
+
+
+        return {"predictions": dict1}
+
+
+if __name__ == "__main__":
+    model = KFServingSampleModel("sales-application")
+    model.load()
+    kfserving.KFServer(workers=1).start([model])
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/model.py:52]: nit: newline

File: docs/samples/custom/timeseries-holtwinter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+ENV APP_HOME /app
+WORKDIR $APP_HOME
+
+# Install production dependencies.
+COPY requirements.txt ./
+RUN pip3 install --no-cache-dir -r ./requirements.txt
+
+# Copy local code to container image
+COPY model.py  model_holt.sav   ./
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/Dockerfile:13]: not sure if it is a good practice to store model in the image  but I guess it is fine for an example.

File: docs/samples/custom/timeseries-holtwinter/model.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        weeks = int(inputs[0]["image"]["weeks"])
+        print(weeks)
+
+        #Loading the joblib files 
+        self.holt =  joblib.load('model_holt.sav')
+        
+
+
+        #creating return dictionary 
+        dict1={"sales":"application"}
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/model.py:37]: Ideally we'd like to make predict generic and application specific processing can be moved to `preprocess` or `postprocess` handler.

File: docs/samples/custom/timeseries-holtwinter/model.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        
+        inputs = request["instances"]
+
+        
+        weeks = int(inputs[0]["image"]["weeks"])
+        print(weeks)
+
+        #Loading the joblib files 
+        self.holt =  joblib.load('model_holt.sav')
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/model.py:32]: This code should move to load as you do not want to load the model for every request.

File: docs/samples/custom/timeseries-holtwinter/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+joblib
+numpy
+pandas
+statsmodels==0.11.1
+kfserving==0.3.0
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/requirements.txt:5]: should use 0.4.0 now

File: docs/samples/custom/timeseries-holtwinter/model.py (hunk 3 above)
Comment (narennadig) [reply docs/samples/custom/timeseries-holtwinter/model.py:37]: @yuzisun  Let me change it

File: docs/samples/custom/timeseries-holtwinter/requirements.txt (hunk 5 above)
Comment (narennadig) [reply docs/samples/custom/timeseries-holtwinter/requirements.txt:5]: @yuzisun  sure

File: docs/samples/custom/timeseries-holtwinter/model.py (hunk 1 above)
Comment (narennadig) [reply docs/samples/custom/timeseries-holtwinter/model.py:52]: @yuzisun  didn't get this reference..

File: docs/samples/custom/timeseries-holtwinter/model.py (hunk 4 above)
Comment (narennadig) [reply docs/samples/custom/timeseries-holtwinter/model.py:32]: @yuzisun okay

File: docs/samples/custom/timeseries-holtwinter/model.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        self.ready = True
+
+    def predict(self, request: Dict) -> Dict:
+        inputs = request["instances"]
+        weeks = int(inputs[0]["image"]["weeks"])
+        print(weeks)
+        #creating  dictionary to return
+        dict1={}
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/model.py:26]: give a better variable name?

Comment (narennadig) [reply docs/samples/custom/timeseries-holtwinter/model.py:26]: @yuzisun  sure

**Comment by yuzisun** [comment]
@narennadig sorry for the delay. Mostly lgtm, just in general we do not want to checkin the actual model file into KFServing repo, is there a public http url where you can download this model instead?

**Comment by narennadig** [comment]
Hey @yuzisun ..there isn't any place as such unfortunately :( ...That's why I added a code to generate it ....

**Comment by yuzisun** [comment]
> Hey @yuzisun ..there isn't any place as such unfortunately :( ...That's why I added a code to generate it ....

I can help upload the model to KFServing example GCS bucket

**Comment by narennadig** [comment]
@yuzisun sure...Let me know when we can connect

File: docs/samples/custom/timeseries-holtwinter/custom.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: sales-application
+spec:
+  default:
+    predictor:
+      custom:
+        name: custom
+        container:
+          image: {docker-username}/kfserving-custom-example
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/custom.yaml:13]: now you can add the storage uri env to download the model automatically to /mnt/model/model_holt.sav
```
apiVersion: serving.kubeflow.org/v1alpha2
kind: InferenceService
metadata:
  labels:
    controller-tools.k8s.io: "1.0"
  name: sales-application
spec:
  default:
    predictor:
      custom:
        name: custom
        container:
          image: {docker-username}/kfserving-custom-example
          env:
          - name: STORAGE_URI
             value: gs://kfserving-examples/models/statsmodel
```

File: docs/samples/custom/timeseries-holtwinter/model.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+class KFServingSampleModel(kfserving.KFModel):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.name = name
+        self.ready = False
+
+    def load(self):
+        #Loading the joblib files 
+        self.model =  joblib.load('model_holt.sav')
```
Comment (yuzisun) [inline docs/samples/custom/timeseries-holtwinter/model.py:18]: you can now load from `/mnt/model/model_holt.sav`

**Comment by yuzisun** [comment]
close stale PR

**Pull request closed without merging**